### ISSUE TITLE
operator raise error want to expand pvc but storageClass doesn't support it, need to handle this case manually

### ISSUE THREAD

**Issue by linuxduduniao** [VictoriaMetrics/operator#821]
* Operator Version:  v0.39.0
* Kubernetes:  v1.28.3-eks-4f4795d
* vmcluster yaml:
```yaml
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMCluster
metadata:
  name: vmcluster
  namespace: monitoring
spec:
  clusterVersion: v1.93.4-cluster
  retentionPeriod: "12"
  replicationFactor: 2       
  vmstorage:
    replicaCount: 3
    storageDataPath: "/vm-data"
    storage:
      volumeClaimTemplate:
        spec:
          resources:
            requests:
              storage: 200Gi
    resources:
      requests:
        cpu: "1"
        memory: 4Gi
      limits:
        cpu: "2"
        memory: 6Gi
  vmselect:
    replicaCount: 2
    cacheMountPath: "/select-cache"
    extraArgs:
      search.maxUniqueTimeseries: "3000000"
      search.maxSamplesPerQuery: "10000000000"
      search.maxQueryLen: "65536"
    storage:
      volumeClaimTemplate:
        spec:
          resources:
            requests:
              storage: 10Gi
    resources:
      requests:
        cpu: "1"
        memory: "500Mi"
      limits:
        cpu: "1"
        memory: "2000Mi"
  vminsert:
    replicaCount: 2
    resources:
      limits:
        cpu: "1"
        memory: "1000Mi"
```
* operator error logs:
```text
{"level":"error","ts":"2023-12-11T06:44:02.329Z","logger":"client_utils","msg":"want to expand pvc but storageClass doesn't support it, need to handle this case manually","pvc":"vmstorage-db-vmstorage-vmcluster-1","stacktrace":"github.com/go-logr/logr.Logger.Error\n\tgithub.com/go-logr/logr@v1.2.4/logr.go:299\ngithub.com/VictoriaMetrics/operator/controllers/factory/k8stools.growSTSPVC\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/k8stools/expansion.go:165\ngithub.com/VictoriaMetrics/operator/controllers/factory/k8stools.HandleSTSUpdate\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/k8stools/sts.go:83\ngithub.com/VictoriaMetrics/operator/controllers/factory.createOrUpdateVMStorage\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/vmcluster.go:294\ngithub.com/VictoriaMetrics/operator/controllers/factory.CreateOrUpdateVMCluster\n\tgithub.com/VictoriaMetrics/operator/controllers/factory/vmcluster.go:63\ngithub.com/VictoriaMetrics/operator/controllers.(*VMClusterReconciler).Reconcile\n\tgithub.com/VictoriaMetrics/operator/controllers/vmcluster_controller.go:77\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:121\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:320\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:234"}
```

**Comment by Haleygo** [comment]
Are you trying to increase storage size in `vmstorage.storage.volumeClaimTemplate.spec.resources.storage`?
operator will report this error when you try to increase the size and the storageClass under it doesn't support [resize](https://github.com/VictoriaMetrics/operator/blob/master/controllers/factory/k8stools/expansion.go#L189).

**Comment by jicki** [comment]
metoo

```
{"level":"error","ts":1702452176.854926,"logger":"client_utils","msg":"want to expand pvc but storageClass doesn't support it, need to handle this case manually","pvc":
```

**Comment by Haleygo** [comment]
> metoo operator version v0.39.4
> 
> Setting WATCH_NAMESPACE to a specific namespace reports this error.
> 
> Setting WATCH_NAMESPACE to an empty value does not report this error.
> 
> ```
> {"level":"error","ts":1702452176.854926,"logger":"client_utils","msg":"want to expand pvc but storageClass doesn't support it, need to handle this case manually","pvc":
> ```

@jicki Yeah, that's on purpose. When single namespace is been specified, operator doesn't have enough permission to list storageClass clusterwide, so the error will be reported when trying to grow storage size in vmstorage.storage.volumeClaimTemplate.spec.resources.storage.
https://github.com/VictoriaMetrics/operator/blob/134141147207487310328ea2fca74ff8dfac4da2/controllers/factory/k8stools/expansion.go#L194-L199

**Comment by f41gh7** [comment]
Hello, it's a limitation of single namespace mode for operator as @Haleygo mentioned.

 Current work around for it - add specifal annotation for persistentVolumeClaimTemplate: `operator.victoriametrics.com/pvc-allow-volume-expansion: "true"`


```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMCluster
...
spec:
  ....
  vmstorage:
   ....
    storage:
      volumeClaimTemplate:
       metadata:
         annotations:
              operator.victoriametrics.com/pvc-allow-volume-expansion: "true"
        spec:
          resources:
            requests:
              storage: 200Gi
```

 Exists persistent volumes must be manually annotated.

**Comment by f41gh7** [comment]
Must be properly fixed at [v0.42.0 release](https://github.com/VictoriaMetrics/operator/releases/tag/v0.42.0)
